pkg/pool/operations/import.go

```diff
@@ -36,14 +36,30 @@ import (
 // - If any error occurred during import operation
 func (oc *OperationsConfig) Import(cspi *cstor.CStorPoolInstance) (bool, error) {
 	if poolExist := checkIfPoolPresent(PoolName(), oc.zcmdExecutor); poolExist {
+		// If the pool is renamed and imported but the pool-mgmt restarts
+		// for some reason then the annotation should be removed.
+		delete(cspi.Annotations, types.OpenEBSCStorExistingPoolName)
 		return true, nil
 	}
 
+	var poolImported, importable bool
+	// existingPoolName denotes the pool name that may be present
+	// from previous version and needs to be imported with new name
+	existingPoolName := cspi.Annotations[types.OpenEBSCStorExistingPoolName]
+	if existingPoolName != "" {
+		_, importable, _ = oc.checkIfPoolIsImportable(cspi)
```

Comment:
do we need to handle the return err here ?

Reply:
No if err occurs it will fail to import and re-attempt.

Reply:
Good to handle but it might not be mandatory.

Reply:
that's correct , but its better to fail fast